docs/rfcs/1005-realm-key-rotation.md

```diff
@@ -0,0 +1,924 @@
+            // `EntryName` encrypted with the key
+            "name": "name",
+            "type": "Bytes"
+        }
+    ]
+}
+```
+
+The certificate might be valid while containing invalid data:
+
+- the `name` field cannot be decrypted by the key
+- the decrypted content of `name` is not a valid `EntryName`
+
+In this case, the naming fallback to the realm ID.
+
+> **Note:**
+>
+> This certificate can only be uploaded with `realm_rename` API command, though
+> `realm_create` may seem appropriated to also have it.
+> However this is not the case (i.e. realm creation an realm name certificate upload
+> are not one atomic operation), this is for three reasons:
+>
+> - It simplifies code.
+> - User realm doesn't need realm name, hence this certificate would need to be optional
+>   or set to a dummy value.
+> - This allows legacy (i.e. realm created, but no realm name certificate) to be treated
+>   just like the regular case.
+
+### 4.5 - New document `RealmKeysBundle`
+
+```json5
+{
+    "label": "RealmKeysBundle",
+    "type": "realm_keys_bundle",
+    "other_fields": [
+        {
+            // Author must be equal to the one in the `RealmRoleCertificate`
+            "name": "author",
+            "type": "DeviceID",
+        },
+        {
+            // Timestamp must be equal to the one in the `RealmRoleCertificate`
+            "name": "timestamp",
+            "type": "DateTime"
+        },
+        {
+            "name": "realm_id",
+            "type": "VlobID"
+        },
+        {
+            "name": "keys",
+            // Keys ordered by index
+            "type": "List<Bytes>"
+        }
+    ]
+}
+```
+
+> **Note:**
+>
+> It could be tempting not to sign the `RealmKeysBundle`, given the keys can be
+> checked against their corresponding key rotation certificate.
+>
+> However this doesn't take into account that a key that doesn't match it key
+> rotation certificate must be kept in the bundle nevertheless (as removing
+> it would mess the key index for the subsequent key rotations).
+>
+> So if the client encounter a realm keys bundle containing an invalid key it
+> cannot tell if it's normal (i.e. due to a previous invalid key rotation), or
+> an attack by the server (no signature means the server can craft a keys bundle
+> with the right number of keys, but with all of them invalid)
+
+### 4.6 - New document `RealmKeysBundleAccess`
+
+```json5
+{
+    "label": "RealmKeysBundleAccess",
+    "type": "realm_keys_bundle_access",
+    "other_fields": [
+        {
+            "name": "keys_bundle_key",
+            "type": "SecretKey"
+        }
+    ]
+}
+```
+
+> **Note:**
+>
+> `RealmKeysBundleAccess` is not signed, this is because it only role it to give
+> access to `RealmKeysBundle`.
+>
+> Hence forging a `RealmKeysBundleAccess` is possible for the server, but only
+> prevent from accessing `RealmKeysBundle` which also can be done by just returning
+> invalid data in place of the actual encrypted `RealmKeysBundle` data.
```

Comment:
> If the server can send the `SecretKey` that mean it can decode the key-bundle or the SecretKey is encrypted by some means ?

user's public key is available to the server, hence the latter can send encrypted payload to the former.

So the server can generate a dummy `SecretKey`, encrypt it for alice, and pretend this is the genuine key bundle access.

Of course the server doesn't know the real `SecretKey` that encrypt the key bundle, so this trick can only lead to Alice not being able to decrypt the key bundle, which can already be achieved by the server by just sending dummy data instead of the genuine encrypted key bundle